python/cudf/cudf/tests/test_parquet.py

```diff
@@ -313,6 +332,10 @@ def test_parquet_read_row_group(tmpdir, pdf, row_group_size):
         pdf = pdf.drop(columns=["col_category"])
     if "col_category" in gdf.columns:
         gdf = gdf.drop("col_category")
+    if "col_str" in pdf.columns:
+        pdf = pdf.drop(columns=["col_str"])
+    if "col_str" in gdf.columns:
+        gdf = gdf.drop("col_str")
```

Comment:
Why are we dropping these?

Reply:
So there seems to be an existing issue around `categories` & `str` when using `cudf` engine for `read_parquet`: https://github.com/rapidsai/cudf/issues/4566

Reply:
This is only for `category` and not for `str`. We shouldn't need to drop `str` columns.

Reply:
So, the issue was around empty dataframe inputs missing all columns, which is now being handled for all dtypes, added tests under `test_write_read_cudf`

Reply:
Is this a bug on the libcudf side of not returning empty columns?

Reply:
This is a current limitation of the reader as the schema is part of the row group in parquet, since we have 0 rows incase of an empty dataframe, there will be zero rowgroups being written which will imply no schema being read by the writer(libcudf).

But pandas actually doesn't have this issue because it seems to be fetching that information from `pandas metadata`. 

So for example, 
(The cudf examples are before the current bug fix..)
```python
>>> df = cudf.DataFrame({'a':cudf.Series([], dtype='float64')})
>>> df
Empty DataFrame
Columns: [a]
Index: []
>>> df.to_parquet('gpu.parquet')
>>> cudf.read_parquet('gpu.parquet')
_metadata->row_groups.size(): 0.  # debug prints 
column_types.size(): 0            # debug prints 
_selected_columns.size(): 0       # debug prints 
Empty DataFrame
Columns: []
Index: []


>>> pd.read_parquet('gpu.parquet')
Empty DataFrame
Columns: [a]
Index: []
```

After having a discussion with @OlivierNV offline regarding the above behavior of the reader, decided to go ahead with handling on the python side easily.

Reply:
Awesome, thanks @galipremsagar!